### PR TITLE
NR-35139: Add ability to set custom meta titles and descriptions

### PR DESCRIPTION
**Pull request by mickeyryan42**
https://issues.newrelic.com/browse/NR-35139

The `description` prop doesn't seem to be used in `IOSeo.js`.  Can we remove it?

I am depending on the slug being the first part of the url which appears to work.  Please let me know if theres a case where this isn't true.

I did some reading on character limits for meta title/descriptions. It seems the limit for title is ~70 and description ~160. However these are apparently automatically truncated if they are too long, so I did not do any manual truncating of the strings and wanted to leave this open for discussion.

### CODE REVIEW

File: src/data/quickstart-metadata.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "slug": "example",
+    "title": "meta title",
+    "description": "meta description"
+  }
+]
```
Comment (aswanson-nr) [inline src/data/quickstart-metadata.json:7]: To make it easier to look up a quickstart, we could use the slug as the key.
```suggestion
{
  "example": {
    "title": "meta title",
    "description": "meta description"
  }
}
```

File: README.md [hunk 2]
```diff
@@ -72,17 +83,33 @@ Runs every day to get various pages that relate to each quickstart from our vend
...
+
+```
+{
+  "slug": {
+    "title": <title which should be in title meta tag>,
+    "description": <description which should be in description meta tag>
+  }
+}
+```
+
```
Comment (aswanson-nr) [inline README.md:104]: I think we need a little more example-wise here. Maybe something like this:

Custom meta tags can be added for a quickstart in [quickstart-metadata.json](src/data/quickstart-metadata.json)
To add a new set of meta tags for a quickstart, add the following to the `quickstart-metadata.json` file.
```js
{
  "<quickstart slug>": {
    "title": <title which should be in title meta tag>,
    "description": <description which should be in description meta tag>
  }
}
```

For example, to add a custom meta title and description to a quickstart with the slug `cool-qs`:
```js
{
  "cool-qs": {
    "title": "Cool QS",
    "description": "Cool qs description for SEO purposes"
  }
}
```

File: src/data/quickstart-metadata.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "slug": {
+    "title": "meta title",
+    "description": "meta description"
+  }
```
Comment (aswanson-nr) [inline src/data/quickstart-metadata.json:5]: As much as I'd like a placeholder here, I'd rather not end up with this matching a quickstart's slug eventually. I think the readme provides enough information to get started.
```suggestion
```

Comment (mickeyryan42) [reply src/data/quickstart-metadata.json:5]: Yeah that makes sense, updated!

**Review by aswanson-nr — DISMISSED** [review]

**Review by tabathadelane — APPROVED** [review]

**Review by aswanson-nr — APPROVED** [review]
Looks good!

**Pull request merged**